### PR TITLE
[SPARK-27831][SQL][TEST] Move Hive test jars to maven dependency

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

This pr moves Hive test jars(`hive-contrib-0.13.1.jar`, `hive-hcatalog-core-0.13.1.jar`, `hive-contrib-2.3.5.jar` and `hive-hcatalog-core-2.3.5.jar`) to maven dependency.

## How was this patch tested?

Existing test

Please note that this pr need test with `maven` and `sbt`.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Thank you for making the PR again, @wangyum .

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
All test passed,
hadoop-2.7 with sbt: https://github.com/apache/spark/pull/24751#issuecomment-497686977
hadoop-2.7 with maven: https://github.com/apache/spark/pull/24751#issuecomment-497775203
hadoop-3.2 with sbt: https://github.com/apache/spark/pull/24751#issuecomment-497906792
hadoop-3.2 with maven: https://github.com/apache/spark/pull/24751#issuecomment-497925612

**Comment by dongjoon-hyun** [comment]
? 

> hadoop-2.7 with sbt: https://github.com/apache/spark/pull/24751#issuecomment-497686977
> hadoop-2.7 with maven: https://github.com/apache/spark/pull/24751#issuecomment-497775203

`[info] Building Spark using SBT with these arguments:  -Phadoop-2.7 -Pkubernetes -Phive-thriftserver -Pkinesis-asl -Pyarn -Pspark-ganglia-lgpl -Phive -Pmesos test:package streaming-kinesis-asl-assembly/assembly
`
> hadoop-3.2 with sbt: https://github.com/apache/spark/pull/24751#issuecomment-497906792
> hadoop-3.2 with maven: https://github.com/apache/spark/pull/24751#issuecomment-497925612

**Comment by wangyum** [comment]
Sorry. It's should be:
hadoop-2.7 with maven: https://github.com/apache/spark/pull/24751#issuecomment-497686977
hadoop-2.7 with sbt: https://github.com/apache/spark/pull/24751#issuecomment-497775203
hadoop-3.2 with sbt: https://github.com/apache/spark/pull/24751#issuecomment-497906792
hadoop-3.2 with maven: https://github.com/apache/spark/pull/24751#issuecomment-497925612

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you again, @wangyum .
Merged to master.

**Comment by wangyum** [comment]
Hi @felixcheung @dongjoon-hyun It seems we have bug in this PR. `--jars` can't be tested because `hive-contrib-2.3.6.jar` and `hive-hcatalog-core-2.3.6.jar` are already in our classpath.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala [hunk 1]
```diff
@@ -108,8 +108,8 @@ class HiveSparkSubmitSuite
     val unusedJar = TestUtils.createJarWithClasses(Seq.empty)
     val jar1 = TestUtils.createJarWithClasses(Seq("SparkSubmitClassA"))
     val jar2 = TestUtils.createJarWithClasses(Seq("SparkSubmitClassB"))
-    val jar3 = TestHive.getHiveContribJar().getCanonicalPath
-    val jar4 = TestHive.getHiveHcatalogCoreJar().getCanonicalPath
+    val jar3 = HiveTestUtils.getHiveContribJar.getCanonicalPath
+    val jar4 = HiveTestUtils.getHiveHcatalogCoreJar.getCanonicalPath
     val jarsString = Seq(jar1, jar2, jar3, jar4).map(j => j.toString).mkString(",")
```
Comment (wangyum) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSparkSubmitSuite.scala]: It should throw an `AnalysisException`  if we remove jar3 and jar4. But it is successful now:
```
[info]   2019-09-04 22:54:30.372 - stderr> 19/09/04 22:54:30 WARN Master: App app-20190904225430-0000 requires more resource than any of Workers could have.
[info]   2019-09-04 22:54:37.434 - stderr> Exception in thread "main" org.apache.spark.sql.AnalysisException: Can not load class 'org.apache.hadoop.hive.contrib.udaf.example.UDAFExampleMax' when registering the function 'example_max', please make sure it is on the classpath;
[info]   2019-09-04 22:54:37.435 - stderr>
```

**Pull request closed without merging**